src/makeCalculator.js

```diff
@@ -37,7 +37,33 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const res = {
+    result: 0,
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+    operate(callback, number) {
+      this.result = callback(number, this.result);
+
+      return this;
+    },
+    add(number, operator) {
+      return operator + number;
+    },
+    subtract(number, operator) {
+      return operator - number;
+    },
+    divide(number, operator) {
+      return operator / number;
+    },
```

Comment:
avoid dividing by zero. You can throw an error for this case

Reply:
I add error but I don't understand why it is there. there is no limits in task about that. And I am not sure that I do it correct.

Reply:
You did it correct.